File: src/main/java/com/dsm/pick/domains/domain/Activity.java
package com.dsm.pick.domains.domain;

import javax.persistence.*;
import java.time.LocalDate;

@Entity
@Table(name = "activity")
public class Activity {

    @Id
    @Column(name = "date")
    private LocalDate date;

    @Column(name = "schedule")
    private String schedule;

    @ManyToOne
    @JoinColumn(name = "second_floor_teacher_id", referencedColumnName = "id")
    private Teacher secondFloorTeacher;

    @ManyToOne
    @JoinColumn(name = "third_floor_teacher_id", referencedColumnName = "id")
    private Teacher thirdFloorTeacher;

    @ManyToOne
    @JoinColumn(name = "forth_floor_teacher_id", referencedColumnName = "id")
    private Teacher forthFloorTeacher;

    public Activity() {}
    public Activity(LocalDate date, String schedule, Teacher secondFloorTeacher, Teacher thirdFloorTeacher, Teacher forthFloorTeacher) {
        this.date = date;
        this.schedule = schedule;
        this.secondFloorTeacher = secondFloorTeacher;
        this.thirdFloorTeacher = thirdFloorTeacher;
        this.forthFloorTeacher = forthFloorTeacher;
    }

    public LocalDate getDate() {
        return date;
    }

    public void setDate(LocalDate date) {
        this.date = date;
    }

    public String getSchedule() {
        return schedule;
    }

    public void setSchedule(String schedule) {
        this.schedule = schedule;
    }

    public Teacher getSecondFloorTeacher() {
        return secondFloorTeacher;
    }

    public void setSecondFloorTeacher(Teacher secondFloorTeacher) {
        this.secondFloorTeacher = secondFloorTeacher;
    }

    public Teacher getThirdFloorTeacher() {
        return thirdFloorTeacher;
    }

    public void setThirdFloorTeacher(Teacher thirdFloorTeacher) {
        this.thirdFloorTeacher = thirdFloorTeacher;
    }

    public Teacher getForthFloorTeacher() {
        return forthFloorTeacher;
    }

    public void setForthFloorTeacher(Teacher forthFloorTeacher) {
        this.forthFloorTeacher = forthFloorTeacher;
    }
}

File: src/main/kotlin/com/dsm/pick/domain/Activity.kt
package com.dsm.pick.domain

import com.dsm.pick.domain.attribute.Schedule
import com.dsm.pick.domain.converter.ScheduleConverter
import java.time.LocalDate
import javax.persistence.*
import javax.validation.constraints.NotBlank
import javax.validation.constraints.NotNull

@Entity
@Table(name = "activity")
class Activity(

    @Id
    @Column(name = "date")
    @NotNull
    val date: LocalDate,

    @Column(name = "schedule")
    @NotBlank
    @Convert(converter = ScheduleConverter::class)
    val schedule: Schedule,

    @ManyToOne
    @JoinColumn(name = "second_floor_teacher_id", referencedColumnName = "id")
    val secondFloorTeacher: Teacher,

    @ManyToOne
    @JoinColumn(name = "third_floor_teacher_id", referencedColumnName = "id")
    val thirdFloorTeacher: Teacher,

    @ManyToOne
    @JoinColumn(name = "forth_floor_teacher_id", referencedColumnName = "id")
    val forthFloorTeacher: Teacher,
)